### ISSUE TITLE
Segmentation Fault when trying to generate RBI for gems

### ISSUE THREAD

**Issue by igbanam** [Shopify/tapioca#1100]
I have a simple application like so…

in `Gemfile`

```rb
source 'https://rubygems.org'

gem 'sinatra'
gem 'thin'
gem 'sorbet-runtime'
gem 'rack-test'

group :development do
  gem 'sorbet'
  gem 'tapioca', require: false
end
```

in `app.rb`
```rb
# frozen_string_literal: true
# typed: true

require 'sinatra/base'
require 'sorbet-runtime'

extend T::Sig

class App < Sinatra::Application
  get '/' do
    puts "Hello World"
  end
end

App.run!
```

I noticed that Sinatra — as a constant — could not be resolved by Sorbet. So I tried to generate the RBI files so, you know, all will be well. I ran `./bin/tapioca gems --all -V` expecting to generate RBI files for the project; I got this instead

```
Requiring all gems to prepare for compiling...  Done

Removing RBI files of gems that have been removed:

  Nothing to do.

~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/methods.rb:158: [BUG] Segmentation fault at 0x0000000000000010
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [arm64-darwin20]

-- Crash Report log information --------------------------------------------
   See Crash Report log file under the one of following:
     * ~/Library/Logs/DiagnosticReports
     * /Library/Logs/DiagnosticReports
   for more details.
Don't forget to include the above Crash Report log file in bug reports.

-- Control frame information -----------------------------------------------
c:0055 p:---- s:0336 e:000335 CFUNC  :super_method
c:0054 p:0058 s:0332 e:000331 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/methods.rb:158
c:0053 p:0030 s:0326 e:000325 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/methods.rb:65
c:0052 p:0106 s:0312 e:000311 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/methods.rb:49 [FINISH]
c:0051 p:---- s:0307 e:000306 CFUNC  :map
c:0050 p:0008 s:0303 e:000302 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/methods.rb:38 [FINISH]
c:0049 p:---- s:0298 e:000297 CFUNC  :each
c:0048 p:0016 s:0294 e:000293 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/methods.rb:37
c:0047 p:0055 s:0286 e:000285 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/methods.rb:23 [FINISH]
c:0046 p:---- s:0278 e:000277 CFUNC  :bind_call
c:0045 p:0248 s:0272 e:000271 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/sorbet-runtime-0.5.10274/lib/types/private/methods/_methods.rb:272 [FINISH]
c:0044 p:0088 s:0266 e:000265 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/listeners/base.rb:26
c:0043 p:0007 s:0261 e:000260 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/pipeline.rb:188 [FINISH]
c:0042 p:---- s:0257 e:000256 CFUNC  :each
c:0041 p:0006 s:0253 e:000252 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/pipeline.rb:188 [FINISH]
c:0040 p:---- s:0248 e:000247 CFUNC  :bind_call
c:0039 p:0248 s:0242 e:000241 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/sorbet-runtime-0.5.10274/lib/types/private/methods/_methods.rb:272 [FINISH]
c:0038 p:0098 s:0236 e:000235 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/pipeline.rb:153
c:0037 p:0012 s:0231 e:000230 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/gem/pipeline.rb:49 [FINISH]
c:0036 p:---- s:0227 e:000226 CFUNC  :bind_call
c:0035 p:0248 s:0222 e:000221 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/sorbet-runtime-0.5.10274/lib/types/private/methods/_methods.rb:272 [FINISH]
c:0034 p:0117 s:0216 e:000214 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/commands/gem.rb:188 [FINISH]
c:0033 p:---- s:0207 e:000206 CFUNC  :bind_call
c:0032 p:0248 s:0201 e:000200 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/sorbet-runtime-0.5.10274/lib/types/private/methods/_methods.rb:272 [FINISH]
c:0031 p:0006 s:0195 e:000194 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/commands/gem.rb:79
c:0030 p:0021 s:0192 e:000191 METHOD ~/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor/shell/basic.rb:44
c:0029 p:0007 s:0186 e:000185 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/commands/gem.rb:78
c:0028 p:0030 s:0182 e:000181 METHOD ~/.rvm/gems/ruby-3.0.0/gems/parallel-1.22.1/lib/parallel.rb:587
c:0027 p:0014 s:0172 e:000171 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/parallel-1.22.1/lib/parallel.rb:354
c:0026 p:0012 s:0169 e:000168 METHOD ~/.rvm/gems/ruby-3.0.0/gems/parallel-1.22.1/lib/parallel.rb:597
c:0025 p:0039 s:0161 e:000159 METHOD ~/.rvm/gems/ruby-3.0.0/gems/parallel-1.22.1/lib/parallel.rb:353
c:0024 p:0285 s:0148 e:000147 METHOD ~/.rvm/gems/ruby-3.0.0/gems/parallel-1.22.1/lib/parallel.rb:288
c:0023 p:0040 s:0137 e:000136 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/executor.rb:31 [FINISH]
c:0022 p:---- s:0131 e:000130 CFUNC  :bind_call
c:0021 p:0248 s:0126 e:000125 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/sorbet-runtime-0.5.10274/lib/types/private/methods/_methods.rb:272 [FINISH]
c:0020 p:0048 s:0120 E:001f70 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/commands/gem.rb:77 [FINISH]
c:0019 p:---- s:0114 e:000113 CFUNC  :bind_call
c:0018 p:0248 s:0109 e:000108 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/sorbet-runtime-0.5.10274/lib/types/private/methods/_methods.rb:272 [FINISH]
c:0017 p:0272 s:0103 e:000102 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/cli.rb:265
c:0016 p:0006 s:0097 e:000096 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca.rb:18
c:0015 p:0012 s:0094 e:000093 METHOD ~/.rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/rubygems/user_interaction.rb:46
c:0014 p:0044 s:0088 e:000087 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca.rb:17 [FINISH]
c:0013 p:---- s:0082 e:000081 CFUNC  :bind_call
c:0012 p:0248 s:0077 e:000076 BLOCK  ~/.rvm/gems/ruby-3.0.0/gems/sorbet-runtime-0.5.10274/lib/types/private/methods/_methods.rb:272 [FINISH]
c:0011 p:0012 s:0071 E:002258 METHOD ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/lib/tapioca/cli.rb:225
c:0010 p:0054 s:0066 e:000065 METHOD ~/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor/command.rb:27
c:0009 p:0040 s:0058 e:000057 METHOD ~/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor/invocation.rb:127
c:0008 p:0235 s:0051 e:000050 METHOD ~/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor.rb:392
c:0007 p:0062 s:0038 e:000037 METHOD ~/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor/base.rb:485
c:0006 p:0122 s:0031 E:001590 TOP    ~/.rvm/gems/ruby-3.0.0/gems/tapioca-0.9.2/exe/tapioca:23 [FINISH]
c:0005 p:---- s:0028 e:000027 CFUNC  :load
c:0004 p:0112 s:0023 E:0015c0 EVAL   ~/.rvm/gems/ruby-3.0.0/bin/tapioca:23 [FINISH]
c:0003 p:---- s:0018 e:000017 CFUNC  :eval
c:0002 p:0189 s:0011 E:000538 EVAL   ~/.rvm/gems/ruby-3.0.0/bin/ruby_executable_hooks:22 [FINISH]
c:0001 p:0000 s:0003 E:000420 (none) [FINISH]
```

Any ideas?

**Comment by igbanam** [comment]
If anyone needs more information, I'd be happy to divulge

**Comment by rafaelfranca** [comment]
It happened in `c:0055 p:---- s:0336 e:000335 CFUNC  :super_method`, which means it is a Ruby bug. Could you please report to the Ruby issue tracker? Thanks.

**Comment by paracycle** [comment]
Yes, this is a Ruby error and has already been fixed in later versions of Ruby 3.0. Please see this comment for more details: https://github.com/Shopify/tapioca/issues/1061#issuecomment-1191658612
